### PR TITLE
Require TTL header for incoming Subscription events.

### LINKED ISSUE #329: Enforce TTL header on incoming Push publish requests
All incoming push publications should require a TTL header. 
Requests missing the TTL should return a 400 response ("Missing TTL header"). 
The TTL should also be included in the allowed CORS headers.

### PR DESCRIPTION
**Pull request by jrconlin**
Resolves #329 

@bbangert r?

### CODE REVIEW

File: autopush/endpoint.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def _delete_completed(self, response):
 class EndpointHandler(AutoendpointHandler):
     cors_methods = "POST,PUT"
     cors_request_headers = ["content-encoding", "encryption",
-                            "crypto-key",
+                            "crypto-key", "ttl",
```
Comment (kitcambridge) [inline autopush/endpoint.py]: Ooh, good catch!

File: autopush/tests/test_integration.py [hunk 2]
```diff
@@ -215,14 +215,9 @@ def send_notification(self, channel=None, version=None, data=None,
...
-                    headers.update({
-                        'Crypto-Key': headers.get('Crypto-Key') + ';' + ckey
-                    })
-                else:
-                    headers.update({
-                        'Crypto-Key': ckey
-                    })
+                headers.update({
+                    'Crypto-Key': headers.get('Crypto-Key') + ';' + ckey
+                })
```
Comment (jrconlin) [inline autopush/tests/test_integration.py]: Why the reduced code? Apparently, we never call vapid with use_header == False.

Comment (bbangert) [inline autopush/tests/test_integration.py]: Yup, got a warning locally about missing code coverage, so I removed it.

**Comment by bbangert** [comment]
r+, just needs rebase

**Pull request merged**